### PR TITLE
[fetch] Use long command line options for better readability

### PR DESCRIPTION
**Pull request by ypid**
I find it important, specially for security focused projects, to be as easy to review and understand as possible. I think using long command line options helps with this as they better document the code without the need to look them all up.

If you agree, the other scripts could be updated as well.

### CODE REVIEW

File: scripts/fetch [hunk 1]
```diff
@@ -9,22 +9,22 @@ cores=$(nproc)
 manifest_dir="${MANIFEST_DIR?}"
 manifest_repo="${MANIFEST_REPO?}"
 
-mkdir -p "${base_dir}" "${manifest_dir}"
+mkdir --parents "${base_dir}" "${manifest_dir}"
```
Comment (daurnimator) [inline scripts/fetch:12]: `mkdir -p` is very widely known. I'd have to look up what `--parents` means

File: scripts/fetch [hunk 2]
```diff
@@ -9,22 +9,22 @@ cores=$(nproc)
...
 
-rsync -Pav "${manifest_repo}/" "${manifest_dir}/"
+rsync -P --archive --verbose "${manifest_repo}/" "${manifest_dir}/"
```
Comment (daurnimator) [inline scripts/fetch:14]: similarly... I `rsync -av` is a common 'phrase'. I would have to look up the man page to check what `--archive` and `--verbose` do.

File: scripts/fetch [hunk 3]
```diff
@@ -9,22 +9,22 @@ cores=$(nproc)
...
 
 [ -d "${manifest_dir}/.git" ] || {
 	export GIT_WORK_TREE="$manifest_dir"
 	export GIT_DIR="${manifest_dir}/.git"
 	git init
 	git add .
-	git commit -a -m "automated cache commit"
+	git commit --all --message "automated cache commit"
```
Comment (daurnimator) [inline scripts/fetch:21]: this one I don't feel either way about.

File: scripts/fetch [hunk 4]
```diff
@@ -9,22 +9,22 @@ cores=$(nproc)
...
 }
 
 cd "${base_dir}"
 
-repo init -u "${manifest_dir}" -m base.xml
+repo init --manifest-url "${manifest_dir}" --manifest-name base.xml
```
Comment (daurnimator) [inline scripts/fetch:26]: this is a good change. `repo` isn't as common of a tool and I don't find the short args memorable.

File: scripts/fetch (hunk 1 above)
Comment (ypid) [reply scripts/fetch:12]: @daurnimator I understand your point. The question is if we should always go for the long variant for consistency or not. I am open to both in this case.

File: scripts/fetch (hunk 2 above)
Comment (ypid) [reply scripts/fetch:14]: Here I would argue that anyone who was read the `rsync` manpage knows that `-a` and `--archive` are equivalent. But someone who has not read it is probably better of with `--archive`. Same with `--verbose` which is more explicit than `-v`. You would really need to check what `--verbose` does compared to `-v` ;-) ?

File: scripts/fetch (hunk 4 above)
Comment (ypid) [reply scripts/fetch:26]: @daurnimator We have a similar knowledge of the command flags as it seems. The above are known, short or long. The repo command are where I really needed to look up and which was the driving force to use the long variants.

File: scripts/fetch (hunk 2 above)
Comment (daurnimator) [reply scripts/fetch:14]: > You would really need to check what `--verbose` does compared to `-v`

Yes! I didn't actually know `-v` was for verbose in rsync; for some reason its just habit for me to type it.

File: scripts/fetch (hunk 1 above)
Comment (daurnimator) [reply scripts/fetch:12]: I'd argue consistency here hurts understandability.

File: scripts/fetch (hunk 4 above)
Comment (daurnimator) [reply scripts/fetch:26]: I'd be happy to merge these `repo` changes alone for now and debate over the others later?

Comment (ypid) [reply scripts/fetch:26]: Sure, I guess there is no need to go crazy about those changes. Feel free to merge #29. You might as well close this PR (#27) and I will be more conservative with similar changes in the future.

File: scripts/fetch (hunk 1 above)
Comment (lrvick) [reply scripts/fetch:12]: Anyone that does not know what --parents means got to be one of todays lucky 10,000. I don't mind the idea of being verbose in code, and sticking to shorthand day to day.

Makes it easier for people new to this stuff.

File: scripts/fetch (hunk 2 above)
Comment (lrvick) [reply scripts/fetch:14]: Again I feel we should optimize for people new to this stuff, than those of us who have been memorizing shell incantations for 20 years. Verbosity is good for security. Make people easily see options and think about what they do as a standard practice.

**Comment by lrvick** [comment]
Hey @ypid Thanks so much for the contributions and interest! I was not expecting any so soon so I have not been keeping up with this.

I expect to get these all merged over weekend.

**Pull request merged**